ports/libass/CMakeLists.txt

```
@@ -76,6 +74,52 @@ find_path(HARFBUZZ_INCLUDE_DIR
 find_library(FRIBIDI_LIBRARY NAMES libfribidi fribidi)
 find_library(HARFBUZZ_LIBRARY NAMES harfbuzz)
 
+OPTION(ASM "asm optimization for significantly faster libass build")
+
+if(ASM)
+  set(ASFLAGS "-I${CMAKE_CURRENT_SOURCE_DIR}/libass -Dprivate_prefix=ass -DPIC")
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64.*|x86_64.*|AMD64.*")
+    set(BITTYPE 64)
+    set(ASFLAGS "${ASFLAGS} -DARCH_X86_64=1")
+    set(ARCH_X86 1)
+  elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "i686.*|i386.*|x86.*")
+    set(BITTYPE 32)
+    set(ASFLAGS "${ASFLAGS} -DARCH_X86_64=0")
+    set(ARCH_X86 1)
+  else()
+    message(FATAL_ERROR "Assembly optimizations are not yet supported on this architecture ${CMAKE_SYSTEM_PROCESSOR} in cmake build system; disabling.")
+  endif()
+  
+  if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+    set(ASFLAGS "${ASFLAGS} -f win${BITTYPE}")
+    if(BITTYPE EQUAL 32)
+      set(ASFLAGS "${ASFLAGS} -DPREFIX")
+    endif()
+  elseif(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    set(ASFLAGS "${ASFLAGS} -f elf${BITTYPE} -DSTACK_ALIGNMENT=16")
+  elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    set(ASFLAGS "${ASFLAGS} -f macho${BITTYPE} -DPREFIX -DSTACK_ALIGNMENT=16")
+  else()
+    message(FATAL_ERROR "Assembly optimizations are not yet supported for your platform ${CMAKE_SYSTEM_NAME} in cmake build system; disabling.")
+  endif()
+
+  set(CMAKE_ASM_NASM_FLAGS "${CMAKE_ASM_NASM_FLAGS} ${ASFLAGS}")
+endif()
+
+if(ASM)

```

```suggestion
if(ASM)
```
https://github.com/libass/libass/blob/062d4ec598b4fd06f75c0cd6a0f43f5d0939a764/libass/Makefile_library.am#L46

The compiled files for the ARCH architecture are different. Although it is a kind of arm, it uses a completely different A64 instruction set.  I think we can differentiate more upstream. 😬

---

The asm of arm architecture has not been introduced in version 0.17.0. I am mainly based on https://github.com/libass/libass/blob/0.17.0/libass/Makefile_library.am .

---

Sorry, was my fault. We should base on stable release version.